backend/corpora/common/entities/collection.py

```diff
@@ -150,6 +150,81 @@ def to_dict(db_object):
 
         return results
 
+    @classmethod
+    def list_collections(cls, session: Session, visibility: str = None) -> typing.List[dict]:
+        """
+        Get a subset of columns, in dict form, for all Collections with the specified visibility. If visibility is None,
+        return *all* Collections.
+        @param session: the SQLAlchemy session
+        @param visibility: the CollectionVisibility string name
+        @return: a list of dict representations of Collections
+        """
+
+        collection_columns = [
+            "id",
+            "name",
+            "visibility",
+            "tombstone",
+            "contact_name",
+            "contact_email",
+            "curator_name",
+            "revised_at",
+            "created_at",
+            "published_at",
+            "description",
+            "publisher_metadata",
+            "revision_of",
+            "tombstone",
+            "owner",  # Needed for determining view permissions
+        ]
+
+        link_columns = [
+            "link_name",
+            "link_url",
+            "link_type",
+        ]
+
+        dataset_columns = [
+            "id",
+            "curator_tag",
+            "tissue",
+            "assay",
+            "disease",
+            "organism",
+            "tombstone",
+        ]
+
+        filters = []
+        if visibility == CollectionVisibility.PUBLIC.name:
+            filters = [DbCollection.visibility == CollectionVisibility.PUBLIC]
+        elif visibility == CollectionVisibility.PRIVATE.name:
+            filters = [DbCollection.visibility == CollectionVisibility.PRIVATE]
+
+        collections = [r.to_dict() for r in session.query(cls.table).filter(*filters).all()]
+
+        resp_collections = []
+        for collection in collections:
+            resp_collection = cls._copy_columns(collection_columns, collection)
+            for entity_col, cols in (("datasets", dataset_columns), ("links", link_columns)):
+                entities = collection.get(entity_col)
+                resp_collection[entity_col] = [cls._copy_columns(cols, entity) for entity in entities]
+            resp_collections.append(resp_collection)
+
+        return resp_collections
+
+    @staticmethod
+    def _copy_columns(attrs: typing.Iterable, from_obj: dict) -> dict:
+        """
+        Return a subset copy of 'from_obj' which contains only the attributes found as strings in iterable 'attrs'
+        @param attrs: the keys to copy
+        @param from_obj: dict from which to copy key-value pairs
+        @return: copied subset of 'from_obj'
+        """
+        to_obj = {}
+        for a in attrs:
+            to_obj[a] = from_obj[a]
+        return to_obj
+
```

Comment:
this helper method is still needed to do column selection. Prefer affirmative selection to specifying the columns that you want removed (i.e., via `remove_attrs`)